client/src/unifycr-sysio.c

```diff
@@ -917,13 +917,18 @@ ssize_t UNIFYCR_WRAP(read)(int fd, void *buf, size_t count)
     /* check whether we should intercept this file descriptor */
     if (unifycr_intercept_fd(&fd)) {
         /* get pointer to file descriptor structure */
+        unifycr_filemeta_t *meta = unifycr_get_meta_from_fid(fd);
         unifycr_fd_t *filedesc = unifycr_get_filedesc_from_fd(fd);
-        if (filedesc == NULL) {
+
+        if (meta == NULL || filedesc == NULL) {
             /* ERROR: invalid file descriptor */
             errno = EBADF;
             return (ssize_t) (-1);
         }
 
+        if (filedesc->pos == meta->size)
```

Comment:
Modified the pread test code to try a normal read() after an lseek with pos past the end of the file, this should return 0 (success) as well, so I think this check should be:

```suggestion
        if (filedesc->pos >= meta->size)
```